geonode/base/api/serializers.py

```diff
@@ -543,8 +543,70 @@ def to_representation(self, instance):
         for lnk in links:
             formatted_link = model_to_dict(lnk, fields=link_fields)
             ret.append(formatted_link)
+            if lnk.asset:
+                extras = {
+                    "type": "asset",
+                    "content": model_to_dict(lnk.asset, ["title", "description", "type", "created"]),
+                }
+                extras["content"]["download_url"] = asset_handler_registry.get_handler(lnk.asset).create_download_url(
+                    lnk.asset
+                )
+                formatted_link["extras"] = extras
+
+        if download_links := self._get_download_links(instance):
+            ret.extend(download_links)
+
         return ret
 
+    class Flopper:
+        status = True
+
+        def get(self):
+            ret = self.status
+            self.status = False
+            return ret
+
+    def _get_download_links(self, instance_id):
```

Comment:
@etj let's keep the downloads out of this PR, since I want to review them a bit. I don't like the spaghetti code we're getting for docs, datasets and 3dtiles.

Reply:
@giohappy edited #12435 to remove download info